compiler/passes/src/code_generation/visit_program.rs

```diff
@@ -167,7 +166,7 @@ impl<'a> CodeGenerator<'a> {
         output_string
     }
 
-    fn visit_function(&mut self, function: &'a Function) -> String {
+    fn visit_function_with(&mut self, function: &'a Function, mut futures: &[Location]) -> String {
```

Comment:
Just a heads up, in the future, we may want to unify code generation under the `Visitor` pattern. This change is fine though.